src/index.js

```diff
@@ -1,3 +1,107 @@
 'use strict'
 
-exports = module.exports = require('./block-service.js')
+const async = require('async')
+
+// BlockService is a hybrid block datastore. It stores data in a local
+// datastore and may retrieve data from a remote Exchange.
+// It uses an internal `datastore.Datastore` instance to store values.
+module.exports = class BlockService {
+  constructor (ipfsRepo) {
+    this._repo = ipfsRepo
+    this._bitswap = null
+  }
+
+  goOnline (bitswap) {
+    this._bitswap = bitswap
+  }
+
+  goOffline () {
+    this._bitswap = null
+  }
+
+  isOnline () {
+    return this._bitswap != null
+  }
+
+  addBlock (block, extension, callback) {
+    if (this.isOnline()) {
+      if (typeof extension === 'function') {
+        callback = extension
+        extension = undefined
+      }
+
+      this._bitswap.hasBlock(block, callback)
+    } else {
+      this._repo.datastore.put(block, extension, callback)
+    }
+  }
+
+  addBlocks (blocks, callback) {
+    if (!Array.isArray(blocks)) {
+      return callback(new Error('expects an array of Blocks'))
+    }
+
+    async.eachLimit(blocks, 100, (block, next) => {
+      this.addBlock(block, next)
+    }, callback)
+  }
+
+  getBlock (key, extension, callback) {
+    if (this.isOnline()) {
+      if (typeof extension === 'function') {
+        callback = extension
+        extension = undefined
+      }
+
+      this._bitswap.getBlock(key, callback)
```

Comment:
So, bitswap will check the repo if it already has the block? (Thought that logic would be here, but having bitswap doing the job is also fine)